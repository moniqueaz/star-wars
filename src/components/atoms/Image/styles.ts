import styled, { css } from 'styled-components';
type ImageProps = {
  size: 'normal' | 'full'
}

const getSize = ( {
  normal: css`width: auto`,
  full: css`width: 100%`,
} );

export const Image = styled.img<ImageProps>`
  ${( { size } ) => css`
    ${getSize[size]}
  `}
`;
