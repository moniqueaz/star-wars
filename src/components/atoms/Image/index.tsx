import * as S from './styles';

type ImageProps = {
  src: string,
  size?: 'normal' | 'full'
}
const Image = ( { src, size = 'normal' }: ImageProps) => {
  if (!src) {
    return null;
  }
  return (
    <S.Image src={src} size={size}/>
  );
};

export default Image;
