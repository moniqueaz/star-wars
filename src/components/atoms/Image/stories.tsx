import { Story, Meta } from '@storybook/react/types-6-0';
import Image from '.';

export default {
  title: 'Atoms/Image',
  component: Image,
} as Meta;

export const Default: Story = args => {
  return <Image {...args} src="https://via.placeholder.com/150"/>;
};
export const Normal: Story = () => {
  return <Image src="https://via.placeholder.com/500x200"/>;
};

export const Full: Story = () => {
  return <Image src="https://via.placeholder.com/500x200" size="full"/>;
};
