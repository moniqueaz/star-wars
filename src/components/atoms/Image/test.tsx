import { render } from '@testing-library/react';

import Image from '.';

describe('<Image />', () => {
  it('should render', () => {
    const { container } = render(<Image src="/"/>);
    expect(container.querySelector('img')).toBeInTheDocument();
    expect(container.querySelector('img')).toHaveStyle( { width: 'auto' } );

    expect(container.firstChild).toMatchSnapshot();
  } );

  it('should render without src', () => {
    const { container } = render(<Image src=""/>);
    expect(container.querySelector('img')).not.toBeInTheDocument();
  } );

  it('should render without full', () => {
    const { container } = render(<Image src="/" size="full"/>);
    expect(container.querySelector('img')).toHaveStyle( { width: '100%' } );
  } );
} );
