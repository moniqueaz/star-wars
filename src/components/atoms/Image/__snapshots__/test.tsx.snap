// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render 1`] = `
.c0 {
  width: auto;
}

<img
  class="c0"
  src="/"
/>
`;
